components/butterfly/src/rumor/dat_file.rs

```diff
@@ -279,15 +286,16 @@ impl DatFileWriter {
         Ok(total)
     }
 
-    fn write_rumor_store<T, W>(&self, writer: &mut W, store: &RumorStore<T>) -> Result<u64>
+    /// # Locking
+    /// * `RumorStore::list` (read) This method must not be called while any RumorStore::list lock
+    ///   is held.
+    fn write_rumor_store_rsr<T, W>(&self, writer: &mut W, store: &RumorStore<T>) -> Result<u64>
         where T: Rumor,
               W: Write
     {
         let mut total = 0;
-        for member in store.list.read().values() {
-            for rumor in member.values() {
-                total += self.write_rumor(writer, rumor)?;
-            }
+        for rumor in store.lock_rsr().rumors() {
```

Comment:
It wasn't immediately obvious when reading the definition for `rumors()` above, but seeing this in action makes me realize how much more ergonomic this is.  This is a great improvement.  ✨